test/controllers/api/v2/trends_controller_test.rb

```diff
@@ -0,0 +1,68 @@
+require 'test_helper'
+
+class Api::V2::TrendsControllerTest < ActionController::TestCase
+  foreman_trend_valid_attrs = { :trendable_type => 'Model' }
+  foreman_trend_invalid_attrs = { :trendable_type => 'NotExists' }
+
+  def setup
+    @foreman_trend = FactoryBot.create(:foreman_trend, :trendable_type => 'Environment')
+    @host = FactoryBot.create(:host)
+    FactoryBot.create(:fact_value, :value => '2.6.9',:host => @host,
+                      :fact_name => FactoryBot.create(:fact_name, :name => 'kernelversion'))
+    @fact_os = FactoryBot.create(:fact_value, :value => 'fedora',:host => @host,
+                      :fact_name => FactoryBot.create(:fact_name, :name => 'operatingsystem'))
+    @fact_trend = FactoryBot.create(:fact_trend, :trendable_type => 'FactName')
+    @fact_trend_valid_attrs = { :trendable_type => 'FactName', :trendable_id => @fact_os.fact_name_id.to_s }
+  end
+
+  test "should get index" do
+    get :index, { }
+    assert_response :success
+    assert_not_nil assigns(:trends)
+    trends = ActiveSupport::JSON.decode(@response.body)
+    assert !trends.empty?
+  end
+
+  test "should create a valid foreman trend" do
+    assert_difference('Trend.trend_source.where(:type => "ForemanTrend").count',1) do
+      post :create, { :trend => foreman_trend_valid_attrs}
+      result = ActiveSupport::JSON.decode(@response.body)
+      assert_equal(result["trendable_type"], "Model")
```

Comment:
Please put the expectation before the actual result, `assert_equal` signature is:
> assert_equal( expected, actual, failure_message = nil )